### PR TITLE
extended: retry on update conflicts in deployment tests

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/openshift/origin/issues/9866

@mfojtik ptal

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
@mfojtik ping

**Comment by mfojtik** [comment]
LGTM one nit.

File: pkg/deploy/util/util.go [hunk 1]
```diff
@@ -280,6 +283,35 @@ func GetAvailablePods(pods []api.Pod, minReadySeconds int32) int32 {
...
+type updateConfigFunc func(d *deployapi.DeploymentConfig)
+
+// UpdateConfigWithRetries will try to update a deployment config and ignore any update conflicts.
+func UpdateConfigWithRetries(dn client.DeploymentConfigsNamespacer, namespace, name string, applyUpdate updateConfigFunc) (*deployapi.DeploymentConfig, error) {
+	var (
+		config *deployapi.DeploymentConfig
+		err    error
+	)
+	deploymentConfigs := dn.DeploymentConfigs(namespace)
+	resultErr := wait.Poll(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
```
Comment (kargakis) [inline pkg/deploy/util/util.go]: updated - I also moved the utility out of deploy/util so that we won't depend on the client library in our utilities

**Comment by mfojtik** [comment]
[merge]

**Pull request merged**